### ISSUE TITLE
no_panic attribute applied to function that can panic (SafePatchIterator::next(), I think)

### ISSUE THREAD

**Issue by hrydgard** [snipsco/tract#167]
Linking with tract like this:
```
tract-core = "0.5.0"
tract-onnx = "0.5.0"
```
I'm getting the following linker error on Windows, using Rust 1.37:

```
         lld-link: error: undefined symbol:

          ERROR[no-panic]: detected panic in function `next`

          >>> referenced by libtract_core-36f780e6812f081d.rlib(tract_core-36f780e6812f081d.tract_core.blvpe490-cgu.9.rcgu.o):(_ZN4core3ptr18real_drop_in_place17h27638d8e99f97d5cE)
          >>> referenced by libtract_core-36f780e6812f081d.rlib(tract_core-36f780e6812f081d.tract_core.blvpe490-cgu.9.rcgu.o):(_ZN165_$LT$$LT$tract_core..ops..cnn..patches..SafePatchIterator$u20$as$u20$core..iter..traits..iterator..Iterator$GT$..next..__NoPanic$u20$as$u20$core..ops..drop..Drop$GT$4drop17hdff2e4cd80e88a13E)
```

Seems like no_panic is used to ensure that there are no panics in SafePatchIterator::next(), but one is in fact there. I don't see the possible panic myself, but something may have changed in a new version of Rust?

Can easily clone and patch this out myself, of course, but this seems fishy.

EDIT: Confirmed this is an issue on current master, and that taking out no_panic makes it link correctly.

**Comment by kali** [comment]
Thanks for the report, Windows support is relatively recent, and the CI coverage is still weak... 

Does the command line project build in release mode or does it fail at linking with this error too ?

In the meantime, I'll try to build a few more things in the windows CI.

**Comment by kali** [comment]
https://github.com/dtolnay/no-panic/issues/9

**Comment by kali** [comment]
So the issue is no_panic indeed, when linking without LTO. OS seems to be neutral to this, linux builds behaved the same. I remove no_panic, it was only protecting two functions anyway, and I can still put it back temporarily if needed.

Side bonus, windows CI now check the command line build.

**Comment by kali** [comment]
Merged, released as 0.5.1

**Comment by hrydgard** [comment]
Cool, thanks!
